### DISCUSSION TITLE
Dual Concentric Rotary Encoders

### DISCUSSION THREAD

**Discussion by ghgsrt** [josefadamcik/SofleKeyboard#204] [Q&A]
Does v2.1 support dual concentric encoders (PE11CD)? Sorry if this is a bit of a dumb question, I don't have much experience with stuff like this.

**Comment by JellyTitan** [comment]
Great question! No, dual concentric encoders are not supported. Looking at [the datasheet](https://www.bourns.com/docs/product-datasheets/pec11d.pdf), the footprint is different - the EC11 has 5 pins whereas the PE11CD has 6. 

However - using dual concentric _knobs_ is an easy way to hide the bottom of an EC11 rotary encoder, so you may see keyboards that appear to have a dual concentric encoder, but are only using the knobs.

This[ Sofle Pico case](https://www.printables.com/model/969590-sofle-pico-cloud-case) has the dual concentic knobs on an EC11:
![image](https://github.com/user-attachments/assets/77e291de-8cb1-47f2-921f-737739469b90)

Here you can see the height difference on a sandwich plate case, versus a case that rises to the bottom of the MX keycaps:
![image](https://github.com/user-attachments/assets/e927878a-c1dd-4af1-bf0a-da02c7729e9c)

Most of the hero shots of Sofle's are from above, so you don't really notice the base of the EC11 poking out. Here's a better shot of a single knob showing the encoder base:
![image](https://github.com/user-attachments/assets/a117352f-a14b-49ee-b73b-a345bae7ab8c)


If QMK supports dual concentric knobs, it would be possible to add them. However - the height of the keycaps makes the bottom-most ring sit so low in the keyboard that it would be difficult to use.

**Reply by ghgsrt** [reply]
I really appreciate the thorough response! In that case, would you be able to help me pick out a compatible set of encoders (or at least point me in the right direction for figuring it out myself)? The thing I really care about is the stiffness of the action (or lack thereof). My ideal would be something that feels pretty much identical to the side wheel on an MX Master 3. I tried buying some encoders a while back, thinking I had figured out how to appropriately spec them out, just to find them to be way stiffer than expected, so clearly I had no idea what I was looking for lol.

**Reply by JellyTitan** [reply]
I'm delighted you asked - i just started using the stacked double knobs, and haven't started to spread the good word.

As far as dialing in the feel of the encoder - I'm not sure how to do that. I've tried a bunch of different heights and stems, ordering cheap generics from AliExpress. In my opinion, EC11's are usually perceived as cheap and interchangeable, like diodes, so it's harder to find a nicer quality. I'm actually using an MX Master 3 myself. I don't think I've ever felt an encoder that has detents as gentle as the MX Master. The one time I did notice a nicer feel on an encoder, it was an expensive Bournes EC11 long stem from Mouser. I think there's only a few big name brands out there, and I assume those will have more consistent build quality - but that's only an assumption. 

[20mm EC11's](https://www.aliexpress.us/item/3256802171959870.html) with about 20 detent points, 5-pin, and a push button are pretty common in the keyboard world. I prefer the d-shaft over the star shaft, because you can attach knobs with grub screws. That makes it easier to adjust how high the knob sits on the shaft.

You might try reaching out to [Theramin goat](https://www.theremingoat.com/) - I know he's crazy for switches, he may have some insight on encoders. 

If you do find a selection guide, or even something that feels like the MX Master - please come back and leave a comment!
